app/Post_Types/Diviner_Field/Types/FieldType.php

```diff
@@ -12,4 +12,7 @@ static public function render( $id, $field_label, $helper = '') {
 		return $field;
 	}
 
+	static public function setup() {
```

Comment:
make this `abstract`: `abstract static public function setup();` to force extending fields to implement it; unless the purpose is to have it provide a base no-op method.
If so then add a `//no-op` comment to indicate it.

Reply:
I added the setup function to the interface as a contract and then added a comment to this parent class. By default, it should perform no operations